pyramid_hypernova/tweens.py

```diff
@@ -10,24 +10,11 @@
 
 
 def hypernova_tween_factory(handler, registry):
-    get_batch_url = registry.settings['pyramid_hypernova.get_batch_url']
-
-    plugins = registry.settings.get('pyramid_hypernova.plugins', [])
-    plugin_controller = PluginController(plugins)
-
-    batch_request_factory = registry.settings.get(
-        'pyramid_hypernova.batch_request_factory',
-        BatchRequest,
-    )
-
-    json_encoder = registry.settings.get('pyramid_hypernova.json_encoder', JSONEncoder())
+    registry = registry
 
     def hypernova_tween(request):
-        request.hypernova_batch = batch_request_factory(
-            batch_url=get_batch_url(),
-            plugin_controller=plugin_controller,
-            json_encoder=json_encoder,
-        )
+        request.hypernova_batch = configure_hypernova_batch(registry)
+
         response = handler(request)
 
         hypernova_response = request.hypernova_batch.submit()
```

Comment:
could we use the context manager here now?

Reply:
++